### PR TITLE
docs: add a blog post for the js-ipfs 0.51.0 release

### PR DESCRIPTION
**Pull request by achingbrain**
Adds blog post with release details.

### CODE REVIEW

File: content/post/111-js-ipfs-0-51-0.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+## 🍪 Types
+
+TypeScript has taken the JS world by storm, allowing developers to be alerted when the interface they are coding against is not what they expect and to enable vast amounts of useful tooling like accurate code completion and inline doc viewing.
+
+There have been [several](https://github.com/vishalkuo/types-ipfs) [community](https://github.com/survirtual/types-ipfs) [led](https://github.com/beenotung/typestub-ipfs) [efforts](https://github.com/zabirauf/ipfs-types) to provide type information for `js-IPFS` but since the project moves so quickly it's an uphill struggle to keep compatibility with the changes that get landed in every release.
+
+With `js-IPFS@0.51.0` types are now a first-class citizen within the codebase.  The project has not gone so far as to convert to TypeScript - we do not wish knowledge of TypeScript to become a prerequisite to contributing to `js-IPFS` development, instead all types are defined using [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) comments and are built and validated during development and Continuous Integration.
+
```
Comment (autonome) [inline content/post/111-js-ipfs-0-51-0.md:22]: Optional: Maybe add a note about how it'll stay compatible, eg if code reviews will require, etc. If any requirement been documented, add a pointer to it?

File: content/post/111-js-ipfs-0-51-0.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+<p style="max-width:1000px;margin-left:auto;margin-right:auto;">
+<img src="/111-js-ipfs-0-51-0/types.png">
+</p>
+
+## SECIO Removal
```
Comment (autonome) [inline content/post/111-js-ipfs-0-51-0.md:27]: Is there any impact on devs using js-ipfs? Any actions needed? If so, should summarize here.

File: content/post/111-js-ipfs-0-51-0.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+The [deprecated](https://blog.ipfs.io/2020-08-07-deprecating-secio/) security transport [SECIO](https://github.com/libp2p/specs/blob/master/secio/README.md) has finally been removed in `js-IPFS@0.51.0`.  It's successor, [Noise](https://github.com/libp2p/specs/blob/master/noise/README.md) was first released with `js-IPFS@0.47.0` and is now the default security transport used by libp2p.
+
+`go-IPFS@0.7.0` recently also removed SECIO support after introducing Noise in `go-IPFS@0.6.0`, which means `js-IPFS@0.51.0` and `go-IPFS@0.7.0` can only talk to other nodes running `go-IPFS@0.6.0` or `js-IPFS@0.47.0` or later.
+
+## 📌 Custom DAG Formats
+
+At it's heart IPFS is about files, which means [UnixFS](https://github.com/ipfs/specs/blob/master/UNIXFS.md) and in terms of [IPLD Formats](https://github.com/ipld/interface-ipld-format) means [dag-pb](https://github.com/ipld/js-ipld-dag-pb) and [ipld-raw](https://github.com/ipld/js-ipld-raw).
```
Comment (autonome) [inline content/post/111-js-ipfs-0-51-0.md:35]: s/it's/its/

File: content/post/111-js-ipfs-0-51-0.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+But what if these codecs don't meet your requirements, what if you want to develop a new codec or use less common ones like [dag-jose](https://github.com/ceramicnetwork/js-dag-jose)?
+
+We don't bundle every available codec with the default `js-IPFS` install because some of them pull in large dependency trees which will unnecessarily slow down the install for people who aren't going to use them.
+
+Previously you've been able to use [custom IPLD formats](https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client#ipld-formats) with `ipfs-http-client` and when running `ipfs` as an in-process node as part of your application, but now you can use them in a running daemon as well.
+
+See the new [custom-ipld-formats](https://github.com/ipfs/js-ipfs/tree/master/examples/custom-ipld-formats) example in the `js-IPFS` repo for more information on how to set this up.
+
+## 🍏 ipfs-core
```
Comment (autonome) [inline content/post/111-js-ipfs-0-51-0.md:47]: "ipfs-core - new slimmer IPFS for library lovers" or something descriptive to get readers reading

File: content/post/111-js-ipfs-0-51-0.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+  node.add(...)
+
+  await node.stop()
+}
+```
+
+and you're all set!  This is the start of an attempt to make the codebase more modular and allow for swapping components out and maybe not pulling in everything all the time to enable power users to create a minimal node that only contains the functionality they require.  Watch this space for more!
+
+## CORS is disabled by default
```
Comment (autonome) [inline content/post/111-js-ipfs-0-51-0.md:85]: s/is/is now/ just to be double explicit

**Review by autonome — APPROVED** [review]
👍🏼👍🏼👍🏼 a few misc edits suggested. so much good stuff in this release!

**Review by vasco-santos — APPROVED** [review]
LGTM! Great job 🎉

File: content/post/111-js-ipfs-0-51-0.md (hunk 1 above)
Comment (achingbrain) [reply content/post/111-js-ipfs-0-51-0.md:22]: I've mentioned the new tooling support and that type docs will be required going forwards.

File: content/post/111-js-ipfs-0-51-0.md (hunk 2 above)
Comment (achingbrain) [reply content/post/111-js-ipfs-0-51-0.md:29]: Good shout, I've added a paragraph about upgrading.

File: content/post/111-js-ipfs-0-51-0.md (hunk 4 above)
Comment (achingbrain) [reply content/post/111-js-ipfs-0-51-0.md]: Done

**Pull request merged**